crates/rethnet_evm/benches/util.rs

```diff
@@ -0,0 +1,166 @@
+    pub const STORAGE_SCALES: [u64; 1] = [1];
+}
+
+#[cfg(not(feature = "bench-once"))]
+mod config {
+    const NUM_SCALES: usize = 4;
+    pub const CHECKPOINT_SCALES: [u64; NUM_SCALES] = [1, 5, 10, 20];
+
+    const MAX_CHECKPOINT_SCALE: u64 = CHECKPOINT_SCALES[NUM_SCALES - 1];
+    pub const ADDRESS_SCALES: [u64; NUM_SCALES] = [
+        MAX_CHECKPOINT_SCALE * 5,
+        MAX_CHECKPOINT_SCALE * 25,
+        MAX_CHECKPOINT_SCALE * 50,
+        MAX_CHECKPOINT_SCALE * 100,
+    ];
+
+    pub const STORAGE_SCALES: [u64; 4] = [1, 10, 100, 1000];
+}
+
+use config::*;
+
+pub enum VaryStorageSlots {
+    Yes,
+    No,
+}
+
+pub fn bench_sync_state_method<O, R, Prep>(
+    c: &mut Criterion,
+    method_name: &str,
+    mut prep: Prep,
+    mut method_invocation: R,
+    vary_storage_slots: VaryStorageSlots,
+) where
+    R: FnMut(Box<dyn SyncState<StateError>>, u64) -> O,
+    Prep: FnMut(&mut dyn SyncState<StateError>, u64),
+{
+    let mut group = c.benchmark_group(method_name);
+    for accounts_per_checkpoint in CHECKPOINT_SCALES.iter() {
+        for number_of_accounts in ADDRESS_SCALES.iter() {
+            for storage_slots_per_account in match vary_storage_slots {
+                VaryStorageSlots::Yes => STORAGE_SCALES.to_vec(),
+                VaryStorageSlots::No => [1].to_vec(),
+            }
+            .iter()
+            {
+                let mut rethnet_states = RethnetStates::default();
+                rethnet_states.fill(
+                    *number_of_accounts,
+                    *accounts_per_checkpoint,
+                    *storage_slots_per_account,
+                );
+
+                for (label, state_factory) in rethnet_states.make_clone_factories().into_iter() {
+                    group.bench_with_input(
+                        BenchmarkId::new(
+                            format!(
+                                "{},{} account(s) per checkpoint, {} storage slots per account",
+                                label, *accounts_per_checkpoint, *storage_slots_per_account
+                            ),
+                            *number_of_accounts,
+                        ),
+                        number_of_accounts,
+                        |b, number_of_accounts| {
+                            b.iter_batched(
+                                || {
+                                    let mut state = state_factory();
+                                    prep(&mut state, *number_of_accounts);
+                                    state
+                                },
+                                |state| method_invocation(state, *number_of_accounts),
+                                BatchSize::SmallInput,
+                            );
+                        },
+                    );
+                }
+            }
+        }
+    }
+}
+
+pub fn prep_no_op(_s: &mut dyn SyncState<StateError>, _i: u64) {}
+
+pub fn _account_has_code(state: &dyn SyncState<StateError>, address: &Address) -> bool {
```

Comment:
```suggestion
pub fn account_has_code(state: &dyn SyncState<StateError>, address: &Address) -> bool {
```

In rust (and other languages like Typescript, for that matter), prepending `_` implies that the function or variable is never used and you're trying to avoid compiler errors/warnings.

Reply:
Indeed. I was coerced by `cargo check` into naming it like this. The issue is that this function is only ever called from within `debug_assert!()`, so when it's built for `release` this function is considered "dead".

Reply:
In this case, that would make it valid to add this on the line above the function:

`#[allow(dead_code)]`

to prevent that warning from occurring.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3910/commits/564ace53efe9849f6d50d54e771f3c18544fac4c